service/share/share.go

```diff
@@ -0,0 +1,124 @@
+package share
+
+import (
+	"context"
+
+	format "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+
+	"github.com/celestiaorg/nmt/namespace"
+
+	"github.com/celestiaorg/celestia-node/ipld"
+	"github.com/celestiaorg/celestia-node/ipld/plugin"
+	"github.com/celestiaorg/celestia-node/service/header"
+)
+
+// TODO(@Wondertan): We prefix real data of shares with namespaces to be able to recover them during erasure coding
+//  recovery. However, that is storage and bandwidth overhead(8 bytes per each share) which we can avoid by getting
+//  namespaces from CIDs stored in IPLD NMT Nodes, instead of encoding namespaces in erasure coding.
+//
+// TODO(@Wondertan): Ideally, Shares structure should be defined in a separate repository with
+//  rsmt2d/nmt/celestia-core/celestia-node importing it. rsmt2d and nmt are already dependent on the notion of "share",
+//  so why shouldn't we have a separated and global type for it to avoid the type mess with defining own share type in
+//  each package.
+//
+// Share is
```

Comment:
Any ideas on how to expand here?

Reply:
Is there a way you can reference how it's defined in NMT repo and then point it at the nmt repo link?

Reply:
Don't get the connection to my original question.

Reply:
If your question is regarding todo, then nmt.PrefixedData == Share

Reply:
@renaynay ?

Reply:
Yes, my comment is just telling you to reference what is in the NMT lib (point users there) for the `Share` doc.

Reply:
@adlerjohn, can you pls expand on what Share is?

Reply:
Could copy from the specs:

> A share is a fixed-size data chunk associated with a namespace ID, whose data will be erasure-coded and committed to in Namespace Merkle trees.

Reply:
Awesome, thanks.